can_read_write.py

```diff
@@ -1,85 +1,40 @@
-# full list of Parameter IDs at https://en.wikipedia.org/wiki/OBD-II_PIDs
-PARAMETER_IDS = {'VEHICLE_BATTERY_PERCENT': 0x2F,
-                 'VEHICLE_SPEED': 0x0D,
-                 }
-
-
-REQUEST_ID = 0x7DF
-REPLY_ID = 0x7E8
-
-
-class FilteredBufferReader(can.BufferedReader):
-    def __init__(self, filter_parameter):
-        """
-        :param filter_parameter: from PARAMETER_IDS
-        """
-        can.BufferedReader.__init__(self)
-        self.buffer = queue.Queue(0)
-        self.filter_parameter = filter_parameter
-
-    def on_message_received(self, msg):
-        if msg.arbitration_id == REPLY_ID and msg.data[2] == self.filter_parameter:
-            self.buffer.put(msg)
-
-
-# establish interface w CAN BUS
-try:
-    bus = can.interface.Bus(channel='can0', bustype='socketcan_native')
-
-except OSError:
-    print('Cannot find PiCan board')
-    sys.exit()
+from flask import Flask
+import serial
 
-# BUS readers filter and queue messages for reading
-speed_reader = FilteredBufferReader([PARAMETER_IDS['VEHICLE_SPEED']])
-battery_percent_reader = FilteredBufferReader(PARAMETER_IDS['VEHICLE_BATTERY_PERCENT'])
+# info handled specified by https://docs.google.com/document/d/1kUU54jQZAB9nwCM-iA96Kj0fXJ6RNw9nAcBffzji5cU/edit
 
-# notifier distributes messages to readers
-notifier = can.Notifier(bus,
-                        listeners=[speed_reader, battery_percent_reader],
-                        timeout=5.0)
+stats = {
+    "soc": 0,
+    "temp": 0
+}
 
 app = Flask(__name__)
-
-
-@app.route('/get-speed')
-def get_speed():
-    bus.send(can.Message(arbitration_id=REQUEST_ID,
-             data=[0x02, 0x01, PARAMETER_IDS['VEHICLE_SPEED'], 0x00, 0x00, 0x00, 0x00, 0x00],
-             extended_id=False))
-
-    rec_msg = speed_reader.get_message()  # returns None if timeout
-
-    # for debug, TODO: remove for production
-    if rec_msg:
-        speed = rec_msg.data[3] / 1.609  # km/h to mi/h
-        return str(speed)
-    else:
-        return str(-1)
-
-
-@app.route('/get-battery-percent')
-def get_battery_percent():
-    bus.send(can.Message(arbitration_id=REQUEST_ID,
-                         data=[0x02, 0x01, PARAMETER_IDS['VEHICLE_BATTERY_PERCENT'], 0x00, 0x00, 0x00, 0x00, 0x00]))
-
-    rec_msg = battery_percent_reader.get_message()
-
-    # for debug, TODO: remove for production
-    if rec_msg:
-        return str(rec_msg.data[3])
-    else:
-        return str(-1)
-
+ser = serial.Serial('/dev/ttyACM0', 115200, timeout=3) 
+handler = logging.FileHandler('/home/pi/log/hud.log')
+handler.setLevel(logging.ERROR)
+app.logger.addHandler(handler)
+
+#def read_serial():
+
+@app.route('/get-data')
+def get_data():
+    #read_serial()
+    line = ser.readline().decode("utf-8")
+    app.logger.error(line)
+    if line:
+        line = line.strip()
+        stats["soc"], stats["temp"] = line.split()
+    return json.dumps(stats)
 
 if __name__ == '__main__':
-    app.run()
+    app.run(debug=False, use_reloader=False)
```

Comment:
use_reloader seems like a good thing to me - it's what keeps the driver HUD from crashing if an exception gets thrown.

Reply:
Apparently use_reloader only matters if debug=True. It might be good to turn debug back on to catch those exceptions. I think Srikar changed it when trying to fix threading stuff